### PR TITLE
feat(fieldGroup): allow wrapper & templateOptions

### LINKED ISSUE #486: FieldGroup Wrapper
Take a look at http://stackoverflow.com/questions/32789050/add-a-wrapper-around-a-fieldgroup?noredirect=1#comment53417430_32789050

I am trying to wrap a fieldGroup and it is constantly erroring.  According to Kent, it is not possible to wrap fieldGroups so I am asking that this feature be worked on and added in a future release.

As a workaround...does anyone have a suggestion on how to accomplish what my SO question is about?  

<b>Review of my issue:</b>  I have a service call that is pushing object into my fieldGroup.  I need these items to be grouped/listed.  The below picture is what I am trying to create.  Each of those checkboxes is a directive.  When any of them at checked, a new section of fields are shown and required.

I have all of the checkboxes displaying (and their hidden sections).  But they are all spaced out and not grouped together like they are in this picture.  Hence why I wanted to make a `<ul>` to wrap around them.

riderschecked

### PR DESCRIPTION
**Pull request by benoror**
fixes #486 & #528 

Working example: http://jsbin.com/zaqeke/21/edit?html,js,output

### CODE REVIEW

**Comment by kentcdodds** [comment]
Fantastic! Perfect. Thanks.

**Comment by BarryThePenguin** [comment]
:clap: :tada: :balloon:

**Comment by holtjonathan** [comment]
Woohoo!  Thanks @benoror!!

**Comment by benoror** [comment]
:smile: np

**Comment by holtjonathan** [comment]
Odd question, do you happen to have the old example of how your nested form example worked?  I am curious to see what all was changed.

**Comment by benoror** [comment]
Rollback the JSBin revision number

On Thu, Oct 29, 2015, 12:04 holtjonathan <notifications@github.com> wrote:

> Odd question, do you happen to have the old example of how your nested
> form example worked? I am curious to see what all was changed.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/formly-js/angular-formly/pull/529#issuecomment-152268016>
> .
>
-- 

Sent from my phone (sorry for possible typos)
Benjamin Orozco - benoror.com

File: src/directives/formly-form.test.js [hunk 1]
```diff
@@ -280,6 +280,48 @@ describe('formly-form', () => {
       })
     })
 
+    it(`should be possible to use a wrapper & templateOptions in a fieldGroup`, () => {
+      formlyConfig.setWrapper({
+        name: 'panel',
+        template: `<div class="panel">
+            <div class="heading">
+              Panel Title: {{options.templateOptions.title}}
```
Comment (hrgui) [inline src/directives/formly-form.test.js]: Do we have access to the shortcuts when we set wrappers for field groups? I tried accessing the to variable and options.templateOptions is only the way to go.

Comment (kentcdodds) [inline src/directives/formly-form.test.js]: hmm... You should... Could you create an example showing that you can't?

Comment (hrgui) [inline src/directives/formly-form.test.js]: http://jsbin.com/kutucisiqo/1/edit?html,js,output

The `$scope.to` variable seems to be only set in the `simplifyLife()` (see https://github.com/formly-js/angular-formly/blob/master/src/directives/formly-field.js#L95) variable, which is only set if the field is not a fieldGroup (https://github.com/formly-js/angular-formly/blob/master/src/directives/formly-field.js#L45)

Comment (kentcdodds) [inline src/directives/formly-form.test.js]: Ah, yes, I suppose that we could set up the `to` variable in that if statement as sort of a one-off. Those other defaults probably wouldn't make sense. Do you fancy submitting a PR? Please see the CONTRIBUTING.md to get started :-)

**Pull request merged**